compendium/AsyncWorkService/include/cppmicroservices/asyncworkservice/AsyncWorkService.hpp

```diff
@@ -67,6 +67,18 @@ namespace cppmicroservices
              * with the std::packaged_task<void()> in order to wait on the async task.
              */
             virtual void post(std::packaged_task<void()>&& task) = 0;
+
+            /**
+             * Optional method to wait for all currently executing tasks to finish
+             *
+             * @note If the AsyncWorkService does not implement this function and override,
+             * the call will default to noop.
+             */
+            virtual void
```

Comment:
should this be optional? I would argue that for correct behavior of any implementation that wants to be an `AsyncWorkService`, a "wait" function that waits on all pending and executing tasks is required. what do you think?

Reply:
Perhaps optional (but strongly recommended) until CppMicroServices 4.0, and then it becomes required, as it would be a breaking change for downstream code to react to, in order to build successfully?

Reply:
We have transitioned to required for the time being. We'll have to decide if this is the correct way forward